src/common/serialization/boost_flat_set.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <boost/container/flat_set.hpp>
+#include <boost/serialization/split_free.hpp>
+#include "common/common_types.h"
+
+namespace boost::serialization {
+
+template <class Archive, class T>
+void save(Archive& ar, const boost::container::flat_set<T>& set, const unsigned int file_version) {
+    ar << static_cast<u64>(set.size());
+    for (auto& v : set) {
+        ar << v;
+    }
+}
+
+template <class Archive, class T>
+void load(Archive& ar, boost::container::flat_set<T>& set, const unsigned int file_version) {
+    u64 count{};
+    ar >> count;
+    set.clear();
+    for (auto i = 0; i < count; i++) {
```

Comment:
```suggestion
    for (u64 i = 0; i < count; i++) {
```
Otherwise `i` is actually an `int` here.